### ISSUE TITLE
Add ignoreFeatureNames: {} to unit-blacklist

### ISSUE THREAD

**Issue by zaneriley** [stylelint/stylelint#3006]
Hi team,

Stylelint's made a huge impact on my team, but I'm working to clear up some of the annoyances. Currently, `unit-blacklist` is flagging media queries. Am I doing something wrong here?

> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Unit blacklist isn't ignoring media queries that use `px` unit.

> Which rule, if any, is this issue related to?

`unit-blacklist`

> What CSS is needed to reproduce this issue?

```css
/* should be allowed */
@media screen and (min-width: 500px) {
  
  .my-class {
    /* should be blacklisted */
    min-width: 500px; 
  }
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "unit-blacklist": [
      "px",
      {
        message:
          "Unexpected PX unit. Please use relative unit. (unit-blacklist)",
        ignoreProperties: {
          px: ["/^border/", "/^transform/", "/^media/", "/^@media/"]
        }
      }
    ],
  }
}
```

> Which version of stylelint are you using?

e.g. `^8.2.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

e.g. "CLI with `stylelint **/*.css`"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No. Though I can confirm it happens in `.less`, `.modules.css` as well as vanilla css.

> What did you expect to happen?

`px` unit within `@media` rules to be ignored.

> What actually happened (e.g. what warnings or errors you are getting)?

e.g. "The following warnings were flagged:"

```shell
test.css
 208:31  ✖  Unexpected PX unit. Please use relative unit.               unit-blacklist
 211:16  ✖  Unexpected PX unit. Please use relative unit.               unit-blacklist
```

**Comment by jeddy3** [comment]
@zaneriley Thanks for the issue and for using the template.

> Stylelint's made a huge impact on my team

That's great to hear.

>  but I'm working to clear up some of the annoyances. 

Good stuff. Working out the kinks benefits everyone.

> Unit blacklist isn't ignoring media queries that use px unit.

I believe this rule is behaving correctly. However, I think there's scope to add a new option to address your particular use case.

Right now, there is an optional secondary option to ignore units for specific _properties_ (via `ignoreProperties: {}`). 

We can add a new optional secondary option to ignore units for specific [_feature names_](https://www.w3.org/TR/mediaqueries-4/#mq-features) (via `ignoreFeatureNames: {}`. I think it'll be more flexible to add an ignore option for _feature names_ rather than [at-rule keywords](https://www.w3.org/TR/css3-conditional/#at-media) (like `media`).

For your use-case it would configure it as such:

```
{
  "rules": {
    "unit-blacklist": [
      "px",
      {
        ignoreProperties: {
          px: ["/^border/", "/^transform/",]
        }
        ignoreFeatureNames: {
          px: ["min-width"]
        }
      }
    ],
  }
}
```

@zaneriley  Please consider contributing this option. There's [a section in the Developer Guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#adding-an-option-to-an-existing-rule) on how to get started.

**Comment by lucmerceron** [comment]
Hello guys,

I would love to help, I can see where I need to add the new tests and the option. I also have read the contributing guidelines.

Can I try to make a PR on this ?

**Comment by CAYdenberg** [comment]
@lucmerceron Absolutely. Don't forget the README (for the rule) either. If it helps, you can always create a PR with part of a solution as a WIP (work-in-progress) to show us exactly what you've tried.

**Comment by zaneriley** [comment]
Much appreciated @lucmerceron!

**Comment by jeddy3** [comment]
Done in #3027
